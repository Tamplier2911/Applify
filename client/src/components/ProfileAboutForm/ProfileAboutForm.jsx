// import "./ProfileAboutForm.scss";
import React, { useState, useEffect } from "react";

// redux
import { connect } from "react-redux";
import { createStructuredSelector } from "reselect";
import { selectCurrentLanguage } from "../../redux/lang/lang.selectors";
import { updateUserDataStart } from "../../redux/auth/auth.actions";

// components
import TextInput from "../TextInput/TextInput";
import Button from "../Button/Button";

// JS Rendering CSS
import {
  ProfileAboutFormContainer,
  ProfileAboutFormTitle,
  ProfileAboutFormElement,
} from "./ProfileAboutFormStyles";

// component constants
import profileAboutFormData from "./ProfileAboutFormConstants";

const ProfileAboutForm = ({ about, lang, updateUserDataStart }) => {
  const {
    profileAboutFormTitle,
    profileAboutFormAbout,
    profileAboutFormButton,
  } = profileAboutFormData[lang];

  const [aboutMeObj, setAboutMeObj] = useState({ aboutMe: about });
  const { aboutMe } = aboutMeObj;

  const onInputChange = (e) => {
    const { name, value } = e.target;
    setAboutMeObj({ ...aboutMeObj, [name]: value });
  };

  const onFormSubmit = (e) => {
    e.preventDefault();
    updateUserDataStart(aboutMeObj);
  };

  useEffect(() => {
    setAboutMeObj({ aboutMe: about });
  }, [about]);

  return (
    <ProfileAboutFormContainer animateIn="bounceInLeft" animateOnce={true}>
      <ProfileAboutFormTitle lang={lang}>
        {profileAboutFormTitle}
      </ProfileAboutFormTitle>
      <ProfileAboutFormElement
        autoComplete="off"
        onSubmit={(e) => onFormSubmit(e)}
      >
        <TextInput
          rows="6"
          onInputChange={(e) => onInputChange(e)}
          label={profileAboutFormAbout}
          name="aboutMe"
          value={aboutMe}
          max="700"
          // placeholder="Don't be shy, share your ideas with me!"
          required
        />
        <Button type="submit" value={profileAboutFormButton} />
      </ProfileAboutFormElement>
    </ProfileAboutFormContainer>
  );
};

const mapStateToProps = createStructuredSelector({
  lang: selectCurrentLanguage,
});

export default connect(mapStateToProps, { updateUserDataStart })(
  ProfileAboutForm
);
