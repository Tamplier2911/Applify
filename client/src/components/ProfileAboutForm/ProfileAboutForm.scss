/*

.profile-about {
  display: grid;
  grid-row-gap: 4rem;
  grid-auto-rows: min-content;

  padding: 2rem;
  box-shadow: 0 0.1rem 0.5rem var(--cl-font);

  &__title {
    // ${getFont}
    font-family: var(--font-title);
    font-weight: 400;
    font-size: 1.7rem;

    margin-top: 2rem;
    @media only screen and (max-width: 425px) {
      //   margin-top: 5rem;
    }
  }

  &__element {
    display: grid;
    grid-row-gap: 2rem;
  }
}

*/
