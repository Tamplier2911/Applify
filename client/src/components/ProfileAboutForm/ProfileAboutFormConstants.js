const profileAboutFormData = {
  eng: {
    profileAboutFormTitle: "Tell about yourself:",
    profileAboutFormAbout: "About Me",
    profileAboutFormButton: "Save Changes",
  },
  rus: {
    profileAboutFormTitle: "Расскажите о себе:",
    profileAboutFormAbout: "Обо мне",
    profileAboutFormButton: "Сохранить",
  },
  ukr: {
    profileAboutFormTitle: "Розкажіть про себе:",
    profileAboutFormAbout: "Про мене",
    profileAboutFormButton: "Зберегти",
  },
};

export default profileAboutFormData;
