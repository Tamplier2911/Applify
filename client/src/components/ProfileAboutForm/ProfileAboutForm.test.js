import React from "react";

// component
import ProfileAboutForm from "./ProfileAboutForm";

// utils
import {
  setShallowC,
  findByTestAttr,
  storeFactory,
  checkProps,
} from "../../tests/testUtils";

describe("<ProfileAboutForm />", () => {
  const initialStore = {
    lang: {
      languagePanelHidden: true,
      currentLanguage: "eng",
    },
  };
  const store = storeFactory(initialStore);

  let wrapper;
  beforeEach(() => {
    wrapper = setShallowC(
      <ProfileAboutForm store={store} about="I love Unicorns!" />
    );
  });

  it("renders without an error", () => {
    const profileAaboutForm = findByTestAttr(wrapper, "profile-about-form");
    expect(profileAaboutForm.length).toBe(1);
  });

  it("renders - children profile about form element", () => {
    const profileAaboutFormElement = findByTestAttr(
      wrapper,
      "profile-about-form-element"
    );
    expect(profileAaboutFormElement.length).toBe(1);
  });

  it("recieves expected props", () => {
    const expectedProps = { about: "I love Unicorns!" };
    const err = checkProps(ProfileAboutForm, expectedProps, "props");
    expect(err).toBeUndefined();
  });

  it("matches snapshot", () => {
    const profileAaboutForm = findByTestAttr(wrapper, "profile-about-form");
    expect(profileAaboutForm).toMatchSnapshot();
  });
});
