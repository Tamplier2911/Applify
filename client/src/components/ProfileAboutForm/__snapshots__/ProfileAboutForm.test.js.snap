// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProfileAboutForm /> matches snapshot 1`] = `
<Styled(ScrollAnimation)
  animateIn="bounceInLeft"
  animateOnce={true}
  data-test="profile-about-form"
>
  <styled.h3
    lang="eng"
  >
    Tell about yourself:
  </styled.h3>
  <styled.form
    autoComplete="off"
    data-test="profile-about-form-element"
    onSubmit={[Function]}
  >
    <TextInput
      label="About Me"
      max="700"
      name="aboutMe"
      onInputChange={[Function]}
      required={true}
      rows="6"
      value="I love Unicorns!"
    />
    <Button
      type="submit"
      value="Save Changes"
    />
  </styled.form>
</Styled(ScrollAnimation)>
`;
